extensions/blocks/recurring-payments/edit.jsx

```diff
@@ -381,10 +381,99 @@ class MembershipsButtonEdit extends Component {
 		);
 	};
 
-	render = () => {
+	renderUpgradeNudges = () => {
 		const { notices, postId } = this.props;
 		const { connected, connectURL, products } = this.state;
 
+		return (
+			<>
```

Comment:
this bit of code is a little hard to reason about, particularly when it gets to the last chunk which has five predicates combined. I wonder if splitting it out into separate name methods would make it clearer what is going on, eg. 

```javascript

showStripeNudge = () => {
    if ( ! this.hasUpgradeNudge && ! this.state.shouldUpgrade & this.state.connected === API_STATE_NOTCONNECTED ) {
        return (<StripeNudge
						blockName="recurring-payments"
						postId={ postId }
						stripeConnectUrl={ connectURL }
					/>);
    }
}

// then in returnUpgradeNudges there is just a series of:

{  this.showStripeNudge() }

```

Reply:
This was already existing code, it was just moved. I think we could improve this after this PR?

Reply:
yeh in that case not critical for this PR, could be a later refactor